modules/python/test/test_misc.py

```diff
@@ -816,6 +816,106 @@ def test_named_arguments_with_output_argument(self):
         np.testing.assert_equal(dst, src_copy)
         self.assertEqual(arguments_dump, 'lambda=25, sigma=5.5')
 
+    def test_arithm_op_with_scalar_issue_24057_regression(self):
```

Comment:
I've prepared a bit more comprehensive test. 

You can manually apply changes from attached patch file:
[arithm_op_python_tests.txt](https://github.com/opencv/opencv/files/12430618/arithm_op_python_tests.txt)

`git apply arithm_op_python_tests.txt`

Reply:
Thank you very much for your test code !!
I think the new test code is more extensible.

However when I run test, there is a small warning.

```python
   @saturate
   def divide(x, y):
       if not isinstance(y, (int, float)):
           _, max_value = get_limits(x.dtype)
           y[y == 0] = max_value
```

When x = int64, and y = uint8, `y[y==0] = max_value` cannot work well.

```
x =  int64  y=  int64
x =  int64  y=  int64
x =  int64  y=  float64
x =  int64  y=  uint8
/home/kmtr/work/opencv/modules/python/test/test_misc.py:104: DeprecationWarning: NumPy will stop allowing conversion of out-of-bound Python integers to integer arrays.  The conversion of 9223372036854775807 to uint8 will fail in the future.
For the old behavior, usually:
    np.array(value).astype(dtype)`
will give the desired result (the cast overflows).
  y[y == 0] = max_value
x =  int64  y=  float64
.x =  uint8  y=  int64
x =  uint8  y=  int64
x =  uint8  y=  float64
x =  uint8  y=  uint8
x =  uint8  y=  float64
```

I try replace from `x.dtype` to `y.dtype`. If it is incorrect, I'm sorry.

```diff
-          _, max_value = get_limits(x.dtype)
+          _, max_value = get_limits(y.dtype)
```

Reply:
The idea behind:
```python
 _, max_value = get_limits(x.dtype)
 y[y == 0] = max_value
```
is to receive zeros in `x` where `y` is padded with zeros.
Otherwise, we have to deal with `DivisionByZero` warning.
So the proper handling should be a casting `y` to [result type after types promotion by NumPy](https://numpy.org/doc/stable/reference/generated/numpy.result_type.html):
```python
dst_dtype = np.result_type(x, y)
y = np.array(y).astype(dst_dtype)
 _, max_value = get_limits(dst_dtype)
 y[y == 0] = max_value
```

Reply:
Thank you for your comment, I'l try it !